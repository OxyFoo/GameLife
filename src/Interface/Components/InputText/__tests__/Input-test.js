import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Graphical components
import { InputText } from '../index';

describe('[Component] InputText', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<InputText />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should update the input value when typed', () => {
        let [ value, setValue ] = [ '', (v = '') => { value = v } ];
        const { getByTestId } = render(
            <InputText
                value={value}
                onChangeText={setValue}
            />
        );

        const input = getByTestId('textInput');
        fireEvent.changeText(input, 'Hello, World!');
        expect(value).toBe('Hello, World!');
    });

    it('should call a callback function when submitted', () => {
        const onSubmitMock = jest.fn();
        const { getByTestId } = render(
            <InputText onSubmit={onSubmitMock} />
        );

        const input = getByTestId('textInput');
        fireEvent(input, 'submitEditing');
        expect(onSubmitMock).toHaveBeenCalled();
    });
});
