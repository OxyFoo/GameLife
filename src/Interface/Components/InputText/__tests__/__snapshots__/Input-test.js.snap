// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] InputText renders correctly 1`] = `
<View
  collapsable={false}
  onLayout={[Function]}
  style={
    {
      "backgroundColor": "#0E1247",
      "borderColor": "#808080",
      "borderRadius": 4,
      "borderWidth": 1.2,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#0E1247",
        "height": 3,
        "left": 6,
        "position": "absolute",
        "top": -2,
        "transform": [
          {
            "scaleX": 0,
          },
        ],
        "width": 0,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "left": 0,
        "overflow": "scroll",
        "paddingHorizontal": 4,
        "position": "absolute",
        "top": 0,
        "transform": [
          {
            "translateX": -0,
          },
          {
            "translateY": -0,
          },
          {
            "scale": 1,
          },
          {
            "translateX": 0,
          },
          {
            "translateY": 0,
          },
          {
            "translateX": 16,
          },
          {
            "translateY": -2,
          },
        ],
      }
    }
  >
    <Text
      onLayout={[Function]}
      style={
        [
          {
            "color": "#ECECEC",
            "fontFamily": "Hind Vadodara",
            "fontSize": 16,
            "margin": 0,
            "padding": 0,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      Default
    </Text>
  </View>
  <TextInput
    autoComplete="off"
    autoCorrect={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onSubmitEditing={[Function]}
    selectionColor="#808080"
    style={
      {
        "color": "#FFFFFF",
        "maxHeight": 256,
        "minHeight": 56,
        "paddingHorizontal": 12,
      }
    }
    testID="textInput"
    textContentType="none"
  />
</View>
`;
